src/utils/command.js

```diff
@@ -23,20 +22,19 @@ class BaseCommand extends Command {
     super(...args)
   }
   // Initialize context
-  async init(_projectRoot) {
+  async init() {
     const cwd = argv.cwd || process.cwd()
-    const projectRoot = findRoot(_projectRoot || cwd) // if calling programmatically, can use a supplied root, else in normal CLI context it just uses process.cwd()
     // Grab netlify API token
     const authViaFlag = getAuthArg(argv)
 
     const [token] = this.getConfigToken(authViaFlag)
 
+    const cachedConfig = await this.getConfig(cwd, state, token)
+    const { configPath, config, repositoryRoot: projectRoot } = cachedConfig
+
     // Get site id & build state
     const state = new StateConfig(projectRoot)
```

Comment:
Let's say I have the following directory structure:
```
repo/
  .git/
  site/
    .netlify/state.json
    netlify.toml
```

`repositoryRoot` will evaluate to `repo`, am I correct?
That makes sense.

But `StateConfig` tries to read `<projectRoot>/.netlify/state.json` which will fail.

The CLI seems to link a folder to a site (and not a project), thus I'll need to link both `repo` and `repo/site` folders for that to work.

Looks like we should separate concerns between evaluating the repository root and finding the location of `state.json`

Reply:
This should now work with https://github.com/netlify/build/pull/1581.

Re-writing the PR accordingly right now.